l3_dump.py

```diff
@@ -1,18 +1,27 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
+"""
+Python script to unpack L3 logging file into human-readable trace messages.
+L3: Lightweight Logging Library, Version 0.1
+Date 2023-12-24
+Copyright (c) 2023-2024
+"""
 import sys
 import struct
 import subprocess
 
 if len(sys.argv) != 3:
-    print(f"Usage: {sys.argv[0]} logfile exe")
-    sys.exit(1)
+    print(f"Usage: {sys.argv[0]} <logfile> <program-binary>")
+    sys.exit(0)
 
-section = subprocess.Popen(["readelf", "-x", ".rodata", sys.argv[2]], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
-stdout, stderr = section.communicate()
-rodata_offs = int(stdout.split('\n')[2].split()[0], 0)
+# pylint: disable-next=line-too-long
+with subprocess.Popen(["readelf", "-x", ".rodata", sys.argv[2]], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True) as section:
+    stdout, stderr = section.communicate()
+    rodata_offs = int(stdout.split('\n')[2].split()[0], 0)
+
+# pylint: disable-next=line-too-long
+with subprocess.Popen(["readelf", "-p", ".rodata", sys.argv[2]], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True) as p:
```

Comment:
Rewrite here and on L17 above to enclose statement using `with` was suggested by pylint for reasons of correct resource mgmt.

Another benefit of running pylint.